### PR TITLE
feat: add URL detection and integrity hash parsing to harness package

### PR DESCRIPTION
**Pull request by ggallen**
## Summary
- Add `IsURL`, `IsAbsPath`, `IsRelPath` helpers for classifying resource references
- Add `ParseIntegrityHash` for extracting and validating `#sha256=...` fragments from URLs
- Foundation for ADR-0038 universal harness access — no callers yet

## Test plan
- [x] Unit tests for all functions with edge cases (userinfo, path traversal, hex validation)
- [x] `go test ./internal/harness/...` passes
- [x] `make lint` passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/harness/url.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+// IsRelPath returns true if s is a relative file path (not a URL and not absolute).
+func IsRelPath(s string) bool {
+	return !IsURL(s) && !IsAbsPath(s)
+}
+
+// ParseIntegrityHash extracts the SHA256 hash from a URL fragment (#sha256=...).
+// Returns the URL without the fragment, the hash value, and whether a valid hash was found.
+func ParseIntegrityHash(rawURL string) (cleanURL, hash string, hasHash bool) {
```
[the remark this answers is not in the document]
Comment (ggallen) [reply internal/harness/url.go:44]: Fixed — `ParseIntegrityHash` now normalizes to lowercase via `strings.ToLower` instead of rejecting uppercase hex. Updated godoc and tests. Pushed in 661a33d8.

File: internal/harness/url.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+// IsAbsPath returns true if s is an absolute file path.
+func IsAbsPath(s string) bool {
+	return filepath.IsAbs(s)
+}
+
+// IsRelPath returns true if s is a relative file path (not a URL and not absolute).
+func IsRelPath(s string) bool {
+	return !IsURL(s) && !IsAbsPath(s)
+}
```
[the remark this answers is not in the document]
Comment (ggallen) [reply internal/harness/url.go:39]: Fixed — `IsRelPath` now returns false for empty strings. Updated godoc and test. Pushed in 661a33d8.

**Review by rh-hemartin — APPROVED** [review]
